policy_set.go

```diff
@@ -453,6 +475,42 @@ func (s *policySets) RemoveWorkspaces(ctx context.Context, policySetID string, o
 	return req.Do(ctx, nil)
 }
 
+// AddWorkspaceExclusions adds workspace exclusions to a policy set.
+func (s *policySets) AddWorkspaceExclusions(ctx context.Context, policySetID string, options PolicySetAddWorkspaceExclusionsOptions) error {
+	if !validStringID(&policySetID) {
+		return ErrInvalidPolicySetID
+	}
+	if err := options.valid(); err != nil {
+		return err
+	}
+
+	u := fmt.Sprintf("policy-sets/%s/relationships/workspace-exclusions", url.QueryEscape(policySetID))
+	req, err := s.client.NewRequest("POST", u, options.WorkspaceExclusions)
```

Comment:
Slightly confused by this usage-- wouldn't you use "options" as the body and decorate it with json annotations to create a request body? This appears as thought the request body is expected to be an array... I don't think that is a typical pattern (see allowed-workspaces on the agent-pools endpoint)

The awkward thing with this implementation is that you would not be able to meaningfully add fields to PolicySetAdd/RemoveWorkspaceExclusionsOptions and have them appear in the request body.

Reply:
Thanks! I will update this

Reply:
Seems like it fails the tests and returns 500. I am trying to pass an array of `workspaces` under `workspace-exclusions` similar to the `AddWorkspacesOptions` and `AddProjectOptions`. Please see: https://github.com/hashicorp/go-tfe/blob/hs26gill/TF-7895-Add-delete-policy-sets-relationships-exclusions/policy_set.go#L254-L278

Reply:
I was the reviewer for that change and didn't notice the problem then 😭

Reply:
This issue feels very familiar, though. I need to audit our relationships controllers to understand this better. Let me get back to you

Reply:
It appears I was wrong and this is the dominant pattern! And I can't think of a better way of implementing it, either.